modules/circe/src/main/scala/tethys/circe/ast/CirceSupport.scala

```diff
@@ -0,0 +1,76 @@
+package tethys.circe.ast
+
+import io.circe.{Json, JsonNumber, JsonObject}
+import io.circe.syntax._
+import tethys.readers.{FieldName, ReaderError}
+import tethys.readers.tokens.TokenIterator
+import tethys.{JsonObjectWriter, JsonReader, JsonWriter}
+import tethys.writers.tokens.TokenWriter
+
+trait CirceSupport {
+  implicit lazy val circeJsonObjectWriter: JsonWriter[JsonObject] = new JsonObjectWriter[JsonObject] {
+    def writeValues(value: JsonObject, writer: TokenWriter): Unit =
+      value.toIterable.foreach { case (k, v) => circeJsonWriter.write(k, v, writer) }
+  }
+
+  implicit lazy val circeJsonWriter: JsonWriter[Json] = new JsonWriter[Json] {
+    private class JsonFolder(writer: TokenWriter) extends Json.Folder[Unit] {
+      def onNull: Unit = writer.writeNull()
+      def onBoolean(value: Boolean): Unit = writer.writeBoolean(value)
+      def onNumber(value: JsonNumber): Unit = writer.writeRawJson(value.asJson.noSpaces)
+      def onString(value: String): Unit = writer.writeString(value)
+      def onArray(value: Vector[Json]): Unit = JsonWriter[Vector[Json]].write(value, writer)
+      def onObject(value: JsonObject): Unit = circeJsonObjectWriter.write(value, writer)
+    }
+
+    def write(value: Json, writer: TokenWriter): Unit = value.foldWith(new JsonFolder(writer))
+  }
+
+  implicit lazy val circeJsonObjectReader: JsonReader[JsonObject] = new JsonReader[JsonObject] {
+    def read(it: TokenIterator)(implicit fieldName: FieldName): JsonObject = {
+      if (!it.currentToken().isObjectStart) ReaderError.wrongJson(s"Expected object start but found: ${it.currentToken()}")
+      else {
+        it.next()
+        var builder = JsonObject.empty
```

Comment:
Mb it's better to use Vector.builder to minimize inner vector allocations on `add` operation